futurex_openedx_extensions/dashboard/views.py

```diff
@@ -308,13 +312,8 @@ def _load_query_params(self, request: Any) -> None:
         date_from = request.query_params.get('date_from')
         date_to = request.query_params.get('date_to')
 
-        try:
-            self.date_from = datetime.strptime(date_from, '%Y-%m-%d').date() if date_from else None
-            self.date_to = datetime.strptime(date_to, '%Y-%m-%d').date() if date_to else None
-        except (ValueError, TypeError) as exc:
-            raise ParseError(
-                'Invalid dates. You must provide a valid date_from and date_to formated as YYYY-MM-DD'
-            ) from exc
+        self.date_from = date_str_to_date_obj(date_from, 'date_from')
+        self.date_to = date_str_to_date_obj(date_to, 'date_to')
```

Comment:
Thank you for unifying this @tehreem-sadat , I suggest more django way to do it
```suggestion
        serializer = ReportDateFilterSerializer(data=request.query_params)
        serializer.is_valid(raise_exception=True)

        self.date_from = serializer.validated_data.get('date_from')
        self.date_to = serializer.validated_data.get('date_to')
```

Just create a general serializer and it'll do the conversion and validation
```python
class ReportDateFilterSerializer(serializers.Serializer):
    date_from = serializers.DateField(
        required=False,
        input_formats=['%Y-%m-%d'],
    )
    date_to = serializers.DateField(
        required=False,
        input_formats=['%Y-%m-%d'],
    )
```